api/v1alpha2/bundle_types.go

```diff
@@ -67,6 +67,8 @@ type ImageSource struct {
 	// This should not be used in a production environment.
 	// +optional
 	InsecureSkipTLSVerify bool `json:"insecureSkipTLSVerify,omitempty"`
+	// CertificateData contains the PEM data of the certificate that is to be used for the TLS connection
+	CertificateData string `json:"certificateData,omitempty"`
```

Comment:
Wondering if instead of accepting as a string - should we instead let users references a secret directly here. I understand that this a BD spec, so we could technically get a secret in CE, process it and send it here - but probably processing it in the unpacking code makes it easier, rather than polluting the CE reconciler?

Reply:
I thought about that, but that potentially meant pushing more arguments down into rukpak, and adding more code to rukpak, which will likely change (i.e. be pulled into operator-controller) in the future. We don't know how this interface will change, so I wrote this to accept a PEM which is literally just thrown into another API. This should mean less code to modify when pulling in rukpak. 
I don't really consider it "polluting" the CE reconciler; the same work would have to be done within the reconciler in some way, shape or form.

Reply:
Yeah, totally understand. So basically CE would instead create a spec that could look something like this?:

```
spec:
     connectionConfig:
                 secretRef:
                           name:
                 insecureSkipTLSVerify:
```

We immediately have to make this change in o-c. So why not make the API look consistent here too is what I'm thinking.

Reply:
Again, it requires adding complexity to the function-based API that is not necessary, and since rukpak is going to be pulled into operator controller; we likely won't be using the bundleDeployment API any more, so why add the effort into rukpak?

Reply:
Resolving this based on offline discussion